crates/accelerate/src/sampled_exp_val.rs

```diff
@@ -53,6 +67,41 @@ fn bitstring_expval(dist: &HashMap<String, f64>, mut oper_str: String) -> f64 {
     exp_val / denom
 }
 
+fn bitstring_expval_bitterm(
+    dist: &HashMap<String, f64>,
+    term: &SparseTermView,
+) -> Result<f64, PyErr> {
+    let inds: Vec<usize> = term.indices.iter().map(|x| *x as usize).collect();
+    let n = term.num_qubits as usize;
+    let denom: f64 = fast_sum(&dist.values().copied().collect::<Vec<f64>>());
+    let exp_val: Result<f64, PyErr> = dist
+        .iter()
+        .map(|(bits, val)| {
+            let temp_product: Result<f64, PyErr> =
+                term.bit_terms
+                    .iter()
+                    .enumerate()
+                    .try_fold(1.0, |acc, (idx, bitterm)| {
+                        if bitterm != &BitTerm::Z
+                            && bitterm != &BitTerm::Zero
+                            && bitterm != &BitTerm::One
+                        {
```

Comment:
We could actually check this very efficiently, since we know the binary representations of `Z`, `Zero` and `One` are `0b**_01`: https://github.com/Qiskit/qiskit/blob/bb100f83c76eaecccb9c2888799ea0b2895980a7/crates/quantum_info/src/sparse_observable/mod.rs#L88

So if we'd only have to verify that the last two bits are `01` and get away with a single equality statement 🙂  something like this should probably work
```suggestion
                        // Z-like (diagonal) terms end in 01
                        if bitterm & 0b11 == 0b01 {
```

Reply:
Thanks for the suggestion. Made this into a single equality statement.